### PR TITLE
fix(cli): update kingpin version to fix --retention-period and other time.Duration type flags

### LINKED ISSUE #1972: kopia: error: time: unknown unit "d" in duration for retention period
Atempting to set a retention period using days.  A CLI command like this:  `--retention-period=<number_of_days>d`

I've tried `--retention-period=7d`, and get this error: kopia: error: time: unknown unit "d" in duration "7d"

Setting hours does work, so like `168h`.

Is this a Kopia issue of what unit is accepted or a Wasabi issue?

### PR DESCRIPTION
**Pull request by shlokchaudhari9**
This PR fixes any errors thrown by standard `time` library while parsing string that mentions days and weeks.
Fix was added in upstream kingpin library https://github.com/alecthomas/kingpin/pull/329 and would now be consumed in Kopia with this version bump.

The fix is also validated with an updated unit test.

Fixes #1972

### CODE REVIEW

**Review by julio-lopez — APPROVED** [review]

**Comment by julio-lopez** [comment]
thanks @shlokchaudhari9

**Comment by julio-lopez** [comment]
Merging and iterating.

@shlokchaudhari9 can you submit the additional test case in a separate PR?

File: cli/command_repository_set_parameters_test.go [hunk 1]
```diff
@@ -102,6 +102,14 @@ func (s *formatSpecificTestSuite) TestRepositorySetParametersRetention(t *testin
...
 	require.Contains(t, out, "Blob retention mode:     GOVERNANCE")
 	require.Contains(t, out, "Blob retention period:   25h0m0s")
+
+	// update retention (use days, weeks, nanoseconds)
+	env.RunAndExpectSuccess(t, "repository", "set-parameters", "--retention-mode", blob.Compliance.String(),
+		"--retention-period", "1w2d6h3ns")
+
+	out = env.RunAndExpectSuccess(t, "repository", "status")
+	require.Contains(t, out, "Blob retention mode:     COMPLIANCE")
+	require.Contains(t, out, "Blob retention period:   222h0m0.000000003s")
```
[the remark this answers is not in the document]
Comment (shlokchaudhari9) [reply cli/command_repository_set_parameters_test.go:112]: Sure! :)

**Comment by shlokchaudhari9** [comment]
@julio-lopez Here's the PR for additional test case
https://github.com/kopia/kopia/pull/2057

**Pull request merged**